base/subarray.jl

```diff
@@ -46,7 +46,7 @@ check_parent_index_match(parent, ::NTuple{N, Bool}) where {N} =
     throw(BoundsError(T(parent, indices, offset1, stride1), I))
 
 # This computes the linear indexing compatibility for a given tuple of indices
-viewindexing() = IndexLinear()
```

Comment:
The function `viewindexing()` is never called without an argument in `Base`, so I assume this line was meant as the fallback case for `viewindexing(tail(I))`. Changing the argument to an empty tuple provides the correct linear indexing if all indices are `ScalarIndex`.